js/views/buyWizardVw.js

```diff
@@ -370,28 +369,46 @@ module.exports = baseVw.extend({
   },
 
   displayMap: function(address){
-    "use strict";
-    var addressString = "";
-    //only create new map if address is valid
+    var addressString = "",
+        $currentIframe;
+
+    this.$buyWizardMap.find('iframe.js-iframe-pending, iframe.js-iframe-leaving')
+      .remove();
+    $currentIframe = this.$buyWizardMap.find('iframe');
+    $currentIframe.addClass('blurMore');
+
     if(address && address.street && address.city && address.state && address.postal_code) {
       addressString = address.street + ", " + address.city + ", " + address.state + " " + address.postal_code + " " + address.displayCountry;
-      addressString = encodeURIComponent(addressString);
-      var hideClass = this.hideMap ? "hide" : "";
-      var newMap = '<div class="flexContainer"><iframe class="' + hideClass + ' js-buyWizardMap"' +
-          'width="525" height="350" frameborder="0" style="border:0; margin-top: 0"' +
-          'src="https://www.google.com/maps/embed/v1/place?key=AIzaSyBoWGMeVZpy9qc7H418Jk2Sq2NWedJgp_4&q=' + addressString + '"></iframe></div>';
-      this.$el.find('.js-buyWizardMap').html(newMap);
+    } else {
+      // if address is invalid, we'll create a dummy address for which google maps will show a map of the world
+      addressString = "123 Street" + ", " + "City" + ", " + "State" + " " + "12345" + " " + "Country";      
     }
-  },
 
-  updateMap: function(){
-    var address = [];
-    address.street = $('#buyWizardStreetInput').val();
-    address.city = $('#buyWizardCityInput').val();
-    address.state = $('#buyWizardStateInput').val();
-    address.postal_code = $('#buyWizardPostalInput').val();
+    addressString = encodeURIComponent(addressString);
+    $iFrame = $('<iframe class="js-buyWizardMap js-iframe-pending positionTop" width="525" height="350" frameborder="0" style="border:0; margin-top: 0; height: 262px" />');
+       
+    if ($currentIframe.length) {
+      this.$buyWizardMap.find('.spinner').removeClass('hide');
+      $iFrame.insertBefore($currentIframe);
+    } else {
+      this.$buyWizardMap.find('.flexContainer')
```

Comment:
It would be better to target a class with the prefix js-, so this doesn't break if someone removes the flexContainer class or adds another one to $buyWizardMap.